### PR TITLE
Lockpicking & Pathing

### PR DESCRIPTION
**Pull request by AdamFx990**
I did some of these commits a few months ago. I think these are all the changes:

- Players run to the thing they're trying to interact with

- Doors can't be opened if they're locked

- Some optimisations to hexagon code

- Other miscellaneous optimisations & refactoring

### CODE REVIEW

**Comment by alexeevdv** [comment]
Thanks! Can you please also update changelog file?

File: src/Game/DoorSceneryObject.cpp [hunk 1]
```diff
@@ -36,89 +36,101 @@
...
+}
+
+bool DoorSceneryObject::opened() const
+{
+    return _opened;
+}
+
+void DoorSceneryObject::setOpened(bool value) {
+    // Don't change if door is locked.
+    if (_locked == false)
```
Comment (Leinnan) [inline src/Game/DoorSceneryObject.cpp]: Why don't use `(!_locked)` instead

File: src/PathFinding/Hexagon.cpp [hunk 2]
```diff
@@ -71,68 +60,14 @@ void Hexagon::setPosition(const Point& pos)
...
-void Hexagon::setHeuristic(unsigned int value)
-{
-    _heuristic = value;
-}
-
 bool Hexagon::canWalkThru()
 {
-    bool canWalkThru = true;
+    // Search hex for any blocking objects...
     for (auto object : _objects)
```
Comment (Leinnan) [inline src/PathFinding/Hexagon.cpp]: Maybe `(const auto& object : _objects)`?

**Comment by alexeevdv** [comment]
Thanks!

**Pull request merged**